accounting/filter.go

```diff
@@ -135,24 +136,27 @@ func preProcessPayments(payments []lndclient.Payment,
 	paymentList := make([]paymentInfo, len(payments))
 
 	for i, payment := range payments {
-		// Try to get our payment destination from our set of htlcs.
-		// If we cannot get it from our htlcs (which is the case for
-		// legacy payments that did not store htlcs), we try to get it
-		// from our payment request. This value may not be present for
-		// all payments, so we do not error if it is not.
-		destination, err := paymentHtlcDestination(payment)
-		if err != nil {
-			destination, err = paymentRequestDestination(
-				payment.PaymentRequest, decode,
-			)
-			if err != nil && err != errNoPaymentRequest {
-				return nil, err
-			}
+		// Attempt to obtain the payment destination and description from our
+		// payment request. If this is not possible (which can be the case for
+		// legacy payments that did not store payment requests, or payments that
+		// pay directly to a payment hash), then try to get it from our HTLCs.
+		// Note that HTLCs may also not be available for legacy payments that
+		// did not store HTLCs.
+		destination, description, err := paymentRequestDetails(
+			payment.PaymentRequest, decode,
+		)
+		if err != nil && err != errNoPaymentRequest {
+			return nil, err
+		}
+
+		if destination == nil {
+			destination, _ = paymentHtlcDestination(payment)
```

Comment:
I think this change alters the preference. Before, the data from `paymentHtlcDestination` took preference over `paymentRequestDestination()`. Now it's the other way around.

Reply:
We need to inspect the invoice in order to get the description, so given that we will always be inspecting the invoice, it seems that inspecting the HTLCs only makes sense as a fallback in case we weren't able to get the destination from the payment request.

Is there an expectation that the destination obtained from the HTLC's could be different from the destination in the payment request?

Reply:
Actually, I'm not sure. That's why I'm not really comfortable changing the logic here. Would need to dig into the lnd code and docs to find out. But maybe @carlaKC has a quick answer for me?

Reply:
I've pushed an additional commit which returns the original order of precedence for the destination

https://github.com/lightninglabs/faraday/pull/182/commits/c478467268092ebea1efc38f497ae115015f315f

It does add some additional overhead though since in many cases we'd be inspecting HTLCs to find the destination even though we already know it from inspecting  the payment request.

Reply:
Could you please squash the two commits?